### PR TITLE
Adds skill and eval templates for contributions

### PR DESCRIPTION
**Pull request by hyd628**
Summary                                                   

  - Merged template/README.md content into the root README.md contributing      
  section, then removed template/README.md to avoid duplication
  - Expanded contributing guide with quick start commands, a checklist, eval    
  requirements, and tips                                                        
  - Made evals a requirement for all new skills — PRs without evals will not be
  merged                                                                        
  - Added a "Skill Content" section to template/SKILL.md with scaffolding for
  key concepts, rules, and common mistakes                                      
  - Added template/evals/evals.json with two example entries (basic use case +
  edge case) as a starting point for contributors

### CODE REVIEW

File: template/evals/evals.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[
+  {
+    "id": "your-skill-basic",
+    "prompt": "A realistic prompt a user would give the agent",
```
Comment (jessiemongeon1) [inline template/evals/evals.json:4]:
```suggestion
    "prompt": "A realistic prompt a user would give the agent",
    "sources": [
        "link-to-docs.com (description of what to get from those docs)",
        "docs.sui.io/references/framework/* (sui, std, bridge, system framework API reference)"
    ],
```

Comment (jessiemongeon1) [reply template/evals/evals.json:4]: "sources" vs "source-constraint":
source_constraint = "the whole skill must only use these docs", 
sources = "here's exactly where to fact-check this specific eval prompt."

File: template/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  different from general-purpose patterns?
+-->
+
+Brief description of what this skill covers and why it exists. Call out the most
+common AI-coding mistakes this skill prevents (e.g., using deprecated APIs,
+confusing Sui with other chains, misunderstanding the object model).
+
+This skill routes to focused reference files. Load only the ones relevant to the
+current task.
+
```
Comment (jessiemongeon1) [inline template/SKILL.md:26]: Here we can add something like: 

```
All patterns in this skill are derived from:
  https://sdk.mystenlabs.com/sui

If unsure about any API, fetch the relevant page before answering.
Do not guess or extrapolate from other SDKs.
```

Comment (jessiemongeon1) [reply template/SKILL.md:26]: but make it general for the template

File: README.md [hunk 3]
```diff
@@ -50,12 +72,51 @@ description: What this skill does and when to use it. Be specific — this is wh
...
+Every new skill must include evals. Evals verify that the skill produces correct, reliable output and prevent regressions as the skill evolves. PRs without evals will not be merged.
+
+Place an `evals/evals.json` file in your skill directory:
+
+```json
+[
+  {
+    "id": "your-skill-basic",
+    "prompt": "A realistic prompt a user would give the agent",
+    "expected_output": "Description of what correct output looks like",
```
Comment (jessiemongeon1) [inline README.md:103]:
```suggestion
    "sources": [
        "https://source-1.com",
        "https://source-2.com"
        ],
    "expected_output": "Description of what correct output looks like",
```

File: README.md [hunk 4]
```diff
@@ -50,12 +72,51 @@ description: What this skill does and when to use it. Be specific — this is wh
...
+    ]
+  }
+]
+```
+
+When writing evals:
+- Cover the core use cases your skill is designed to handle
+- Use realistic prompts that match how a user would actually invoke the skill
+- Include edge cases where the skill's domain has common pitfalls
+- Each eval should test a distinct capability — avoid redundant prompts
```
Comment (jessiemongeon1) [inline README.md:116]:
```suggestion
- Each eval should test a distinct capability — avoid redundant prompts
- Each prompt should include a `sources` field that links out to documentation that should be used to verify the output of the eval. Sources can be set for individual prompts, or if the entire eval pulls from the same source, you can use the `source_constraint` heading at the top of the eval file. 
```

**Comment by hyd628** [comment]
@jessiemongeon1 feedback has been incorporated. Ready for review again.

**Review by jessiemongeon1 — APPROVED** [review]
